### PR TITLE
(#5489) Groovy > Yaml

### PR DESCRIPTION
**Pull request by tdiesler**
There will be so many changes ... so perhaps we do this incrementally

### CODE REVIEW

**Review by squakez — COMMENTED** [review]
LGTM. However we must not add any file example here directly. For those we have the https://github.com/apache/camel-k-examples/tree/main/generic-examples 

All along the documentation we just need to replace the snippets with the yaml DSL, that should be enough. And yes, feel free to open separate PR and do it iteratively if you prefer.

**Comment by squakez** [comment]
> Do you know what this is ... https://github.com/apache/camel-k/tree/main/docs/modules/ROOT/examples/js
> 
> ```
> $ git grep ck.js                                    
> docs/antora.yml:    requires: "'util=camel-website-util,ck=xref:js/ck.js'"
> ```

No, unfortunately not. We have some configuration magic that I am not familiar with. Mind that certain configuration are required by the main Camel project which is in charge to include the subproject (hence, Camel K among others) bits.

File: docs/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -24,18 +24,23 @@ Users of Camel K can instantly run integration code written in Camel DSL on thei
...
-
-```groovy
-from('timer:tick?period=3000')
-  .setBody().constant('Hello world from Camel K')
-  .to('log:info')
+Just write a _helloworld.yaml_ integration file with the following content:
+
+```yaml
+- from:
+    uri: "timer:yaml"
```
Comment (squakez) [inline docs/modules/ROOT/pages/index.adoc:31]: I'd leave the `timer:tick` instead. Any newcomer may confuse the name of the component with the DSL.

File: docs/modules/ROOT/pages/configuration/build-time-properties.adoc [hunk 2]
```diff
@@ -8,17 +8,21 @@ You may be required to provide certain *build-time properties* that are needed o
...
 
-[source,groovy]
-.build-property-route.groovy
+[source,yaml]
+.build-property-route.yaml
 ----
-from('timer:build-property')
-    .log('The application name: {{quarkus.application.name}}')
+- from:
+    uri: "timer:yaml"
```
Comment (squakez) [inline docs/modules/ROOT/pages/configuration/build-time-properties.adoc:15]: `timer:build-property`

File: docs/modules/ROOT/pages/configuration/build-time-properties.adoc [hunk 3]
```diff
@@ -35,17 +39,21 @@ quarkus.application.name = my-super-application
...
```
Comment (squakez) [inline docs/modules/ROOT/pages/configuration/build-time-properties.adoc:46]: 'timer:build-property'

File: docs/modules/ROOT/pages/configuration/build-time-properties.adoc [hunk 4]
```diff
@@ -63,17 +71,21 @@ kubectl create configmap my-cm-bp --from-literal=quarkus.application.name="my-gr
...
```
Comment (squakez) [inline docs/modules/ROOT/pages/configuration/build-time-properties.adoc:78]: 'timer:build-property'

File: docs/modules/ROOT/pages/configuration/components.adoc [hunk 5]
```diff
@@ -2,19 +2,36 @@
...
-```
+----
 
 For example, you can do it when running the integration from the command line:
 
+[source,yaml]
+.config-seda-route.yaml
+----
+- from:
+    uri: "timer:yaml"
```
Comment (squakez) [inline docs/modules/ROOT/pages/configuration/components.adoc:23]: 'timer:seda'?

File: docs/modules/ROOT/pages/configuration/configuration.adoc [hunk 6]
```diff
@@ -7,17 +7,22 @@ The property value can be used inside Camel K integrations using the *property p
...
 
-[source,groovy]
-.props.groovy
+[source,yaml]
+.property-route.yaml
 ----
-from('timer:props?period=1000')
-    .log('{{my.message}}')
+- from:
+    uri: "timer:yaml"
```
Comment (squakez) [inline docs/modules/ROOT/pages/configuration/configuration.adoc:14]: props

File: docs/modules/ROOT/pages/configuration/runtime-properties.adoc [hunk 7]
```diff
@@ -8,35 +8,45 @@ During the execution of an `Integration` you can provide a single property or a
...
 
-[source,groovy]
-.property-route.groovy
+[source,yaml]
+.property-route.yaml
 ----
-from('timer:property')
-    .log('property content is: {{my.message}}')
+- from:
+    uri: "timer:yaml"
```
Comment (squakez) [inline docs/modules/ROOT/pages/configuration/runtime-properties.adoc:15]: property

**Review by squakez — COMMENTED** [review]
Some minor changes, the rest is good, thanks!

**Review by squakez — APPROVED** [review]

**Comment by tdiesler** [comment]
Ready to merge

**Review by gansheer — APPROVED** [review]

**Comment by tdiesler** [comment]
All docs should be done - ready to merge

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
If we merge, let's run the ci-build for website here: https://ci-builds.apache.org/job/Camel/job/Camel.website/job/main/

in case something is broken we need to fix it soon or revert.

**Comment by tdiesler** [comment]
```
$ ./docs/local-build.sh quick
[15:31:01.130] INFO (@djencks/antora-timer): stage playbookBuilt took 0.167s
[15:31:01.132] INFO (@djencks/antora-timer): Extensions after stage playbookBuilt took 0.002s
[15:31:01.135] INFO (@djencks/antora-timer): stage beforeProcess took 0.003s
[15:31:01.136] INFO (@djencks/antora-timer): Extensions after stage beforeProcess took 0.001s
[15:31:01.229] INFO (@djencks/antora-timer): stage uiLoaded took 0.093s
[15:31:01.229] INFO (@djencks/antora-timer): Extensions after stage uiLoaded took 0s
[15:31:01.258] INFO (@djencks/antora-timer): stage contentAggregated took 0.029s
[15:31:01.259] INFO (@djencks/antora-timer): Extensions after stage contentAggregated took 0.001s
[15:31:01.262] INFO (@djencks/antora-timer): stage contentClassified took 0.003s
[15:31:01.289] INFO (@djencks/antora-timer): Extensions after stage contentClassified took 0.027s
[15:31:02.842] INFO (@djencks/antora-timer): stage documentsConverted took 1.553s
[15:31:02.842] INFO (@djencks/antora-timer): Extensions after stage documentsConverted took 0s
[15:31:02.849] INFO (@djencks/antora-timer): stage navigationBuilt took 0.007s
[15:31:02.849] INFO (@djencks/antora-timer): Extensions after stage navigationBuilt took 0s
[15:31:03.049] INFO (@djencks/antora-timer): stage pagesComposed took 0.2s
[15:31:03.049] INFO (@djencks/antora-timer): Extensions after stage pagesComposed took 0s
[15:31:03.050] INFO (@djencks/antora-timer): stage redirectsProduced took 0.001s
[15:31:03.050] INFO (@djencks/antora-timer): Extensions after stage redirectsProduced took 0s
[15:31:03.052] INFO (@djencks/antora-timer): stage siteMapped took 0.002s
[15:31:03.052] INFO (@djencks/antora-timer): Extensions after stage siteMapped took 0s
[15:31:03.052] INFO (@djencks/antora-timer): stage beforePublish took 0s
[15:31:03.052] INFO (@djencks/antora-timer): Extensions after stage beforePublish took 0s
Site generation complete!
Open file:///Users/tdiesler/git/camel-website/documentation in a browser to view your site.
[15:31:03.133] INFO (@djencks/antora-timer): stage sitePublished took 0.081s
[15:31:03.133] INFO (@djencks/antora-timer): stage sitePublished took 0s
[15:31:03.133] INFO (@djencks/antora-timer): total build time 2.17s
```

**Comment by oscerd** [comment]
> ```
> $ ./docs/local-build.sh quick
> [15:31:01.130] INFO (@djencks/antora-timer): stage playbookBuilt took 0.167s
> [15:31:01.132] INFO (@djencks/antora-timer): Extensions after stage playbookBuilt took 0.002s
> [15:31:01.135] INFO (@djencks/antora-timer): stage beforeProcess took 0.003s
> [15:31:01.136] INFO (@djencks/antora-timer): Extensions after stage beforeProcess took 0.001s
> [15:31:01.229] INFO (@djencks/antora-timer): stage uiLoaded took 0.093s
> [15:31:01.229] INFO (@djencks/antora-timer): Extensions after stage uiLoaded took 0s
> [15:31:01.258] INFO (@djencks/antora-timer): stage contentAggregated took 0.029s
> [15:31:01.259] INFO (@djencks/antora-timer): Extensions after stage contentAggregated took 0.001s
> [15:31:01.262] INFO (@djencks/antora-timer): stage contentClassified took 0.003s
> [15:31:01.289] INFO (@djencks/antora-timer): Extensions after stage contentClassified took 0.027s
> [15:31:02.842] INFO (@djencks/antora-timer): stage documentsConverted took 1.553s
> [15:31:02.842] INFO (@djencks/antora-timer): Extensions after stage documentsConverted took 0s
> [15:31:02.849] INFO (@djencks/antora-timer): stage navigationBuilt took 0.007s
> [15:31:02.849] INFO (@djencks/antora-timer): Extensions after stage navigationBuilt took 0s
> [15:31:03.049] INFO (@djencks/antora-timer): stage pagesComposed took 0.2s
> [15:31:03.049] INFO (@djencks/antora-timer): Extensions after stage pagesComposed took 0s
> [15:31:03.050] INFO (@djencks/antora-timer): stage redirectsProduced took 0.001s
> [15:31:03.050] INFO (@djencks/antora-timer): Extensions after stage redirectsProduced took 0s
> [15:31:03.052] INFO (@djencks/antora-timer): stage siteMapped took 0.002s
> [15:31:03.052] INFO (@djencks/antora-timer): Extensions after stage siteMapped took 0s
> [15:31:03.052] INFO (@djencks/antora-timer): stage beforePublish took 0s
> [15:31:03.052] INFO (@djencks/antora-timer): Extensions after stage beforePublish took 0s
> Site generation complete!
> Open file:///Users/tdiesler/git/camel-website/documentation in a browser to view your site.
> [15:31:03.133] INFO (@djencks/antora-timer): stage sitePublished took 0.081s
> [15:31:03.133] INFO (@djencks/antora-timer): stage sitePublished took 0s
> [15:31:03.133] INFO (@djencks/antora-timer): total build time 2.17s
> ```

This is good, but the full build from the website on the CI might fail anyway. It happened before. But this is great! Thanks @tdiesler

**Comment by tdiesler** [comment]
Could we run this again pls?

**Comment by tdiesler** [comment]
Ready to merge

**Comment by oscerd** [comment]
https://ci-builds.apache.org/job/Camel/job/Camel.website/job/main/1612/

**Comment by tdiesler** [comment]
Thanks @oscerd - the msg for the squashed commit is a little awkward
As you might recall, I squash stuff already before I do a PR

**Comment by oscerd** [comment]
Sorry for that.

**Comment by oscerd** [comment]
The build seems to be fine. The website should be online in a few minutes

**Comment by tdiesler** [comment]
I rebased those changes on main and see no differences. I believe I already changed stuff at the source i.e. the trait Go files

**Comment by squakez** [comment]
See: https://github.com/apache/camel-k/blob/4317da83d0f15137b12da6a2204d2543bd9e4603/docs/modules/traits/pages/3scale.adoc?plain=1#L21

We need to make the change on the go file and then, `make generate`.

**Comment by tdiesler** [comment]
Like so ...
https://github.com/apache/camel-k/pull/5518/commits/787a1d6f826be2a94523ed06588586387be574fe

**Pull request merged**